### DISCUSSION TITLE
Security - HTTP server ? Is the best choice ?

### DISCUSSION THREAD

**Discussion by dmarro89** [dmarro89/dare-db#3] [Ideas]
Is the HTTP server the best choice to interact with the database?
Or should it allow just TCP connection ?

**Comment by vdmitriyev** [comment]
I would suggest to keep both for a while:

1. HTTP/HTTPS
2. Socket based one 

Keeping HTTP/HTTPs interface could be very beneficial, because it quite easy to consume a key-value store available over HTTP/HTTPS from any programming language. It will not be that fast, but for most cases it could be enough. 

On other hand, communicating over sockets to exchange data between application and key-value store could be very fast. However, in this case libraries and packages for other programming languages must be implemented.

**Reply by dmarro89** [reply]
I'll create an issue (for the future) to implement a TCP communication also if it's not urgent (just a reminder for the future).
Thanks

**Comment by dmarro89** [comment]
@vdmitriyev what about grpc or graphql interface to access to the database ?

**Reply by vdmitriyev** [reply]
Good question. I would suggest gRPC as an addition communication channel to existing HTTP/HTTPS. Thus, ``` graphql``` support could be added later on through additional component.

**Comment by go-while** [comment]
1) HTTP/s is perfect but has lots of overhead.

2) (almost) raw TCP would by my first choice
a static/active connection and sending pure (text/bytes/commands) via [dare-db-client module *work_in_progress*](https://github.com/go-while/dare-db-client/blob/main/client/client.go) , if working with go, else goto 1)

3) grapql or grpc, i don't know. never used'em.

**Reply by go-while** [reply]
-> 2) raw tcp can be wrapped around ssl too, simply like httpserver does.

**Reply by go-while** [reply]
https://github.com/go-while/nntp-history/blob/main/SOCKET.go

**Comment by dmarro89** [comment]
Closing this discussion.
Output : 

- HTTP/HTTPS implemented 🆗 
- TCP to be implemented -> https://github.com/dmarro89/dare-db/issues/41
- To evaluate other types of communication -> https://github.com/dmarro89/dare-db/issues/42
